src/main/java/seedu/address/model/person/Birthday.java

```diff
@@ -24,21 +24,25 @@
      * @throws IllegalValueException if given address string is invalid.
      */
     public Birthday(String birthday) throws IllegalValueException {
-        if (!isValidBirthday(birthday)) {
+        String trimmedBirthday = birthday == null ? null : birthday.trim();
+        if (!isValidBirthday(trimmedBirthday)) {
             throw new IllegalValueException(MESSAGE_BIRTHDAY_CONSTRAINTS);
         }
-        this.value = birthday;
+        this.value = trimmedBirthday;
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid date.
      */
-    static boolean isValidBirthday(String birthday) {
+    static boolean isValidBirthday(String test) {
```

Comment:
I feel that using test as a variable is misleading. Maybe use testBday or something instead?  Or just birthday

Reply:
Other than that it seems like your changes are exactly the same as mine except mine has some broken tests :(

Reply:
Good point on `test` variable. I change it for the consistency throughout all `Email.java`, `Address.java` etc. Maybe we should update all of them.

Reply:
The test variable should be consistent. We should update it in the next PR or so.